pyext/src/output.py

```diff
@@ -801,20 +863,96 @@ def get_fields(self, fields, filtertuple=None, filterout=None, get_every=1):
                             continue
 
-                [outdict[field].append(d[self.invstat2_dict[field]])
+                [outdict[field].append(d[self.inv_dict[field]])
                  for field in fields]
         f.close()
         return outdict
 
+    def return_models_satisfying_criteria(self, criteria):
+        # Given a set of criteria, return lines from the stat file that
+        # satisfy all criteria
+        #
+        # Criteria should as a list of tuples in the format: ("TheKeyToBeFiltered",relationship,value)
+        #            where relationship = "<", "==", or ">"
+        #           and keytobefiltered is the name of the key (rather than the integer)
+        # 
+        # Returns a list of dictionaries
+        output_list = []
+        i = 0
+
+        # print fields values
+        f = open(self.filename, "r")
+        line_number = 1
+
+        # skip the first line for a statfile v2
+        if self.isstat2:
+            f.readline()
+
+        for line in f.readlines():
+            append=True
+            fields = self.parse_line(line)
+
+            # Loop over all criteria.  If one fails, the whole line fails and do not append it.
+            for c in criteria:
+                if not self.does_line_pass_criteria(fields, c):
+                    append=False
+                    break
+            if append:
+                output_list.append(fields)
+
+        return output_list
+
+    def _float_string(self, c):
+        # Returns a float if the string can be cast as such.
+        # otherwise, just return the string
+        try:
+            float(c)
+        except:
+            return str(c)
+
+        return float(c)
+
+
+    def does_line_pass_criteria(self, fields, c):
+        # Given a stat file line (as a dictionary) and a criteria tuple, decide whether 
+        # the criteria is passed (return True) or not (return False)
+        
+        key = c[0]
+        if key not in self.get_keys():
+            raise Exception('ERROR: IMP.pmi.output.ProcessOutput.does_line_pass_criteria() - Key %s is not in this stat file' % (key))  
+
+        # Try to cast value string as int, float or, if not, keep it as a string
+        value = self._float_string(c[1])
+
+
+        comparison = c[2]
+        if comparison not in ["==", "<", ">"]:
+            raise Exception('ERROR: IMP.pmi.output.ProcessOutput.does_line_pass_criteria() - Comparison string must be \'>\', \'<\' or \'==\', instead of %s' % (comparison))  
+
+        intkey = self.inv_dict[key]
+
+        model_value = self._float_string(fields[intkey])
+
+        if type(value) is not type(model_value):
+            raise Exception('ERROR: IMP.pmi.output.ProcessOutput.does_line_pass_criteria() - Comparison field %s is of type %s while you tried to compare it to a %s' % (key, type(model_value), type(value)))  
+
+        if (comparison == "==" and model_value == value) or \
```

Comment:
This is unnecessarily verbose. Rather than `if condition: return True; else: return False` just use `return condition`.